packages/connectors/connector-gitlab/src/index.ts

```diff
@@ -0,0 +1,187 @@
+      responseType: 'code',
+      clientId,
+      scope: scopesWithOpenid.join(' '), // Defaults to 'openid' if not provided
+      redirectUri,
+      state,
+    });
+  };
+
+/**
+ * Asynchronously fetches user information from a social media provider.
+ *
+ * @param {GetConnectorConfig} getConfig - A function to get the connector configuration.
+ *
+ * @returns {GetUserInfo} - A function that retrieves user information.
+ *
+ * @throws {ConnectorError} - If an error occurs during the retrieval of user information.
+ */
+const getUserInfo =
+  (getConfig: GetConnectorConfig): GetUserInfo =>
+  async (data, getSession) => {
+    const authResponseResult = oauth2AuthResponseGuard.safeParse(data);
+
+    if (!authResponseResult.success) {
+      throw new ConnectorError(ConnectorErrorCodes.AuthorizationFailed, data);
+    }
+
+    const { code } = authResponseResult.data;
+
+    const config = await getConfig(defaultMetadata.id);
+    validateConfig(config, gitlabConfigGuard);
+
+    const { clientId, clientSecret } = config;
+    const { redirectUri } = await getSession();
+
+    if (!redirectUri) {
+      throw new ConnectorError(ConnectorErrorCodes.General, {
+        message: 'Cannot find `redirectUri` from connector session.',
+      });
+    }
+
+    const tokenResponse = await requestTokenEndpoint({
+      tokenEndpoint,
+      tokenEndpointAuthOptions: {
+        method: TokenEndpointAuthMethod.ClientSecretBasic,
+      },
+      tokenRequestBody: {
+        grantType: 'authorization_code',
+        code,
+        redirectUri,
+        clientId,
+        clientSecret,
+      },
+    });
+
+    const parsedTokenResponse = accessTokenResponseGuard.safeParse(await tokenResponse.json());
+
+    if (!parsedTokenResponse.success) {
+      throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, parsedTokenResponse.error);
+    }
+
+    const { access_token: accessToken, token_type: tokenType } = parsedTokenResponse.data;
+
+    assert(accessToken, new ConnectorError(ConnectorErrorCodes.SocialAuthCodeInvalid));
+
+    try {
+      const userInfoResponse = await ky.get(userInfoEndpoint, {
+        headers: {
+          authorization: `${tokenType} ${accessToken}`,
+        },
+        timeout: defaultTimeout,
+      });
+
+      const rawData = parseJson(await userInfoResponse.text());
+
+      const parsedUserInfoResponse = userInfoResponseGuard.safeParse(rawData);
+
+      if (!parsedUserInfoResponse.success) {
+        throw new ConnectorError(ConnectorErrorCodes.InvalidResponse, parsedUserInfoResponse.error);
+      }
+
+      const {
+        email,
+        email_verified,
+        name,
+        picture: avatar,
+        profile,
+        preferred_username,
+        sub: id,
+      } = parsedUserInfoResponse.data;
+
+      return {
+        id,
+        avatar,
+        email: email_verified ? email : undefined,
+        name,
+        rawData,
+        email_verified,
+        profile,
+        preferred_username,
+      };
```

Comment:
```suggestion
      return {
        id,
        avatar,
        name,
        ...conditional(email_verified && { email }),
        rawData,
      };
```
Can refer to [implementation on Discord connector](https://github.com/logto-io/logto/blob/57d97a4df82c6a5a7132c4cc3b978557b8d5c308/packages/connectors/connector-discord/src/index.ts#L127).

Reply:
HAAAA!!!! Thank you very much for this one!!!!! I was like "I don't really like much the ternary here" hahahaha

Reply:
done